algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+      choices: frequencies,
+      when: answers => answers.interestRateType === "Nominal",
+      filter: capitalizacion => inDays(capitalizacion),
+    },
+    {
+      name: "interestRateFrequency",
+      type: "list",
+      message: "Especifique la frecuencia de la tasa de interés: ",
+      choices: frequencies,
+      filter(interestRateFrequency) {
+        return inDays(interestRateFrequency);
+      },
+    },
+    {
+      name: "interestRate",
+      type: "input",
+      message: "Ingrese el valor de la tasa de interés: ",
+      validate: percentageValidation,
+      filter: value => {
+        const parsed = Number(value);
+        if (percentageValidation(parsed) === true) {
+          return parsed;
+        }
+        return value;
+      },
+      transformer: input => `${input}%`,
+    },
+    {
+      name: "extraCosts",
+      type: "confirm",
+      message: "¿Desea registrar costos extra (iniciales o periódicos)?",
+      default: true,
     },
   ]);
 
-  console.log(`Hello, ${inputs.username}!`);
+  const extraCosts = [];
+
+  if (inputs.extraCosts) {
+    do {
+      const data = await inquirer.prompt([
+        {
+          name: "name",
+          type: "input",
+          message: "Nombre del costo: ",
+          validate: input => input.length > 0,
+        },
+        {
+          name: "type",
+          type: "list",
+          message: "Elija el tipo de gasto: ",
+          choices: ["Inicial", "Periódico"],
+        },
+        {
+          name: "valueType",
+          type: "list",
+          message: "Indique el tipo de valor: ",
+          choices: ["Monetario", "Porcentual"],
+        },
+        {
+          name: "value",
+          type: "input",
+          message: "Ingrese el valor: ",
+          validate: extraCostValidation,
+          filter: (value, answers) => {
+            const parsed = Number(value);
+            if (extraCostValidation(parsed, answers) === true) {
+              return parsed;
+            }
+            return value;
+          },
+          transformer: (value, answers) => {
+            if (answers.valueType === "Monetario") {
+              return currencyFormatter.format(value);
+            }
+            return `${value}%`;
+          },
+        },
+        {
+          name: "keepAdding",
+          type: "confirm",
+          message: "¿Quiere continuar añadiendo costos?",
+          default: true,
+        },
+      ]);
+      extraCosts.push(data);
+    } while (extraCosts[extraCosts.length - 1].keepAdding);
+  }
+
+  const tasaIndicadorBruto = await inquirer.prompt({
+    name: "descuentoKs",
+    type: "input",
+    message: "Tasa de descuento Ks: ",
+    validate: percentageValidation,
+    filter: value => {
+      const parsed = Number(value);
+      if (percentageValidation(parsed) === true) {
+        return parsed;
+      }
+      return value;
+    },
+    transformer: input => `${input}%`,
```

Comment:
Use the percentage formatter everywhere you can.

Reply:
Done